Fixes of HypVINN & options for conform that disable critieria

dkuegler opened the pull request:
- Some fixes to the HypVINN pipeline 
  - formatting/doc
  - naming of HypoVINN->HypVINN (unified)
  - fix some errors with default hypvinn file names and arguments
  - generally improve and clean up the hypvinn interface and functions
- conform.py gets new flags `--no_lia`, `--no_strict_lia`, `--no_img_size`, `--no_vox_size` which deactivate parts of the conform script:
  - `no_lia`: rotation of the affine is not touched, image may remain RAS
  - `no_strict_lia`: rotation of the affine is adapted, but only towards LIA such that no interpolation would be necessary, but will also be LIA (never RAS)
  - `no_img_size` no padding/cropping of the image
  - `no_iso_vox`~`no_vox_size`~ no harmonization of voxel size, for example in combination with `no_strict_lia`, the image is not made iso. voxelsize
-  Cleanup of shell scripts
  -  Remove extra "2&>1" in shell scripts in echo statements (they never have stderr output)

dkuegler committed:
Move documentation utils from advanced into scripts

santiestrada32 committed:
Update paths to save and plot fuction and update type of parser for qc_snap

dkuegler committed:
Fix outdtype of N4bias correction

dkuegler committed:
Add new name for conformed T2
Remove 2&>1 where it is not needed (with echo)

dkuegler committed:
Modify conform.py to support --no_lia, --no_strict_lia, --no_vox_size, --no_img_size

dkuegler committed:
Rename HypoVINN to HypVINN

dkuegler committed:
Fix arguments for hypvinn t2 conforming

m-reuter commented on FastSurferCNN/data_loader/conform.py:
```diff
@@ -107,16 +129,42 @@ def options_parse():
+             "Superceeds --no_strict_lia",
+    )
+    advanced.add_argument(
+        "--no_vox_size",
```
maybe better no_iso_vox ?? or no_vox_iso ?

dkuegler replied:
I guess that is also true and way easier to understand

m-reuter commented on FastSurferCNN/data_loader/conform.py:
```diff
@@ -107,16 +129,42 @@ def options_parse():
+        "--no_strict_lia",
+        dest="force_strict_lia",
+        action="store_false",
+        help="Ignore the forced lia reorientation.",
```
use "LIA" instead of "lia" also in line 154.

m-reuter commented on FastSurferCNN/data_loader/conform.py:
```diff
@@ -107,16 +129,42 @@ def options_parse():
+    )
+    advanced.add_argument(
+        "--no_vox_size",
+        dest="force_vox_size",
```
also here, maybe better force_iso_vox

m-reuter commented on FastSurferCNN/data_loader/conform.py:
```diff
@@ -107,16 +129,42 @@ def options_parse():
+        help="Ignore the forced isometric voxel size (depends on --conform_min).",
+    )
+    advanced.add_argument(
+        "--no_img_size",
```
why not no_img_dim ?

dkuegler replied:
dim is usually the number of dimensions, so usually 3 or 4 in python (numpy/torch)
shape would be what we mean, but I am not sure it that is more "helpful"

m-reuter replied:
yes, I think size is fine

m-reuter commented on FastSurferCNN/data_loader/conform.py:
```diff
@@ -560,33 +661,57 @@ def conform(
+    """
+    _v2v = np.abs(vox2vox[:3, :3])
+    # check 1: have exactly 3 times 1/-1 rest 0, check 2: all 1/-1 or 0
+    return abs(_v2v.sum() - 3) > eps or np.any(np.maximum(_v2v, abs(_v2v - 1)) > eps)
```
shouldn't this be abs(_v2v.sum()) - 3 ? and also abs(_v3v) - 1 ?

dkuegler replied:
v2v is already abs in line 699

m-reuter commented on recon_surf/recon-surf.sh:
```diff
@@ -531,16 +531,16 @@ if [[ ! -f "$mdir/transforms/talairach.lta" ]] || [[ ! -f "$mdir/transforms/tala
   echo " " 2>&1 | tee -a $LF
   echo "============= Computing Talairach Transform ============" 2>&1 | tee -a $LF
   echo " " 2>&1 | tee -a $LF
```
Here (and maybe other places) the 2>&1 can also be removed

dkuegler committed:
Apply changes from Review

- remove more 2>&1
- rename no_vox_size and force_vox_size to no_iso_vox and force_iso_vox
- capitalize LIA
